submissions/Nik3264/oop-exercise/index.js

```diff
@@ -0,0 +1,125 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _https://github.com/Nik3264/OOP-exercise_
+   Web app: _https://nik3264.github.io/OOP-exercise/_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Animal {
+  constructor({ species, name, gender, legs }) {
+    this.species = species || "spesies not assigned";
+    this.name = name || "name not assigned";
+    this.gender = gender || "gender not assigned";
+    this.legs = legs || "legs not assigned, may be 4";
+    this.properties = ["species", "name", "gender", "legs"];
+  }
+
+  getProperties() {
+    return this.properties.map((property) => {
+      return `${property}: <strong>${this[property]}</strong>`;
+    });
+  }
+}
+
+class Dog extends Animal {
+  constructor(name, gender) {
+    super({ name, gender });
+    this.species = "dog";
+    this.legs = 4;
+    Dog.setSaying("gav-gav!");
+  }
+
+  static setSaying(saying) {
+    this._saying = saying;
+  }
+  static getSaying() {
+    return this._saying;
+  }
+  getProperties() {
+    return [
+      ...super.getProperties(),
+      `saying: <strong>${Dog.getSaying()}</strong>`,
+    ];
+  }
+}
+
+class Cat extends Animal {
+  constructor(name, gender) {
+    super({ name, gender });
+    this.species = "cat";
+    this.legs = 4;
+    Cat.setSaying("mio-mi!");
+  }
+
+  static setSaying(saying) {
+    this._saying = saying;
+  }
+  static getSaying() {
+    return this._saying;
+  }
+  getProperties() {
+    return [
+      ...super.getProperties(),
+      `saying: <strong>${Cat.getSaying()}</strong>`,
+    ];
+  }
+}
+
+class Catwoman extends Cat {
+  constructor(name) {
+    super(name);
+    this.species = "catwoman";
+    this.gender = "female";
+    this.legs = 2;
+    this.hands = 2;
+  }
+
+  getProperties() {
+    return [...super.getProperties(), `hands: <strong>${this.hands}</strong>`];
+  }
+}
+
+class Human extends Animal {
+  constructor(name, gender, saying) {
+    super({ name, gender });
+    this.species = "men";
+    this.legs = 2;
+    this.hands = 2;
+    this.saying = saying || "I may saying samething.";
+  }
+  getProperties() {
+    return [
+      ...super.getProperties(),
+      `hands: <strong>${this.hands}</strong>`,
+      `saying: <strong>${this.saying}</strong>`,
+    ];
+  }
+}
+
+function createInstance(classNam, number = 0) {
+  const arrayOfInstans = [];
+  for (let i = 0; i < number; i++) {
+    arrayOfInstans.push(Object.create(classNam));
```

Comment:
const nameOfInstance = new Class();
const cat1=new Cat(‘Murka’, ‘female’); 
const cat2=new Cat(‘Vasya’, ‘male’);
//here you can specify the specific values of the fields of the object instance, and so it was possible to create all the necessary instances for all objects.
But I wanted to automate this process and I created the createInstance function. And how it is possible automatically create many instances so that they have different parameters? This is a kind of incubator where there is only an instance with some default values. But these settings can be changed in the future. Just like 10 people are born, and only then they are given a name.

Reply:
So, why `Object.create`?

>  And how it is possible automatically create many instances so that they have different parameters?

One of the possible approaches:
```
const inhabitants = [
 { classDefinition: Human, individualsProperties: [{name: 'Jamie'}, {name: 'Jason'}], },
 { classDefinition: Dog, individualsProperties: [{name: 'Bo'}, {name: 'Jo'}], },
].map(({classDefinition, individualsProperties}) => individualsProperties.map(individualProperty => new classDefinition(individualProperties))).flat();
```

If `individualProperties` is an array of arrays of properties rather than array of objects then `new classDefinition(individualProperties)` can be `new classDefinition(...individualProperties)`

Reply:
Object.create creates an object based on the prototype, which is not what we want. I was looking for a way how to create an object when the class name is passed through a variable. I didn’t manage to do it right away and I used the switch operator for this. I haven't seen your next comment yet.
And now I'll try to redo the function according to your example, I have a similar solution. I'll just throw out the switch operator and send commit

Reply:
I just sent a commit. 
Your solution is very elegant! And is my decision correct? 
Thank you ))